spec/controllers/api/v2/sections_controller_index_spec.rb

```diff
@@ -0,0 +1,50 @@
+RSpec.describe Api::V2::SectionsController do
```

Comment:
This veers into my personal opinion so take or ignore as appropriate.

Controller specs are deprecated and have been for a while, their suggested replacement is request specs. Could this be a request spec instead?

It is also delving into the structure of the CSV _but_ I'd think the structure of the serialized csv gets covered by the serializer spec, this would just be an additional place to update for any future serialisation changes?

To my mind a request spec cares that 'given expected params does the controller return a csv' (eg check response status, check mime type header?)

Reply:
Well, I have the same opinion - mainly because that's the recommendation of the rails community, but also because controller specs hide a lot of the request/response cycle from the tester. We have a mixture in the code base and I haven't made it an overt goal to actually convert all of our controller specs to request specs. It is a bit of a smell wherever we use `render_views`, too.